### PR TITLE
Remove adsModule feature flag and it's use.

### PR DESCRIPTION
**Pull request by benbowler**
## Summary

Addresses issue:

- #8541

## Relevant technical choices

Additionally to the PR IB, I removed `assets/js/modules/analytics-4/components/common/AdsConversionIDTextField.js` as it is no longer used.

## PR Author Checklist

- [x] My code is tested and passes existing unit tests.
- [ ] My code has an appropriate set of unit tests which all pass.
- [x] My code is backward-compatible with WordPress 5.2 and PHP 7.4.
- [x] My code follows the [WordPress](https://make.wordpress.org/core/handbook/best-practices/coding-standards/) coding standards.
- [ ] My code has proper inline documentation.
- [x] I have added a QA Brief on the issue linked above.
- [x] I have signed the Contributor License Agreement (see <https://cla.developers.google.com/>).

---------------

_Do not alter or remove anything below. The following sections will be managed by moderators only._

## Code Reviewer Checklist

- [ ] Run the code.
- [ ] Ensure the acceptance criteria are satisfied.
- [ ] Reassess the implementation with the IB.
- [ ] Ensure no unrelated changes are included.
- [ ] Ensure CI checks pass.
- [ ] Check Storybook where applicable.
- [ ] Ensure there is a QA Brief.

## Merge Reviewer Checklist

- [ ] Ensure the PR has the correct target branch.
- [ ] Double-check that the PR is okay to be merged.
- [ ] Ensure the corresponding issue has a ZenHub release assigned.
- [ ] Add a changelog message to the issue.

### CODE REVIEW

**Comment by benbowler** [comment]
One note I found in testing is that the Ads Conversion ID is still shown on the Analytics Settings View even after you click "Got it" to confirm the transfer has taken place, this could be confusing IMO because it looks like you don't have a conversion ID set if you came back later and you forgot about the migration process. I suggest we hide this field from the settings view once the message has been dismissed.

**Comment by benbowler** [comment]
Addressed as in the IB, the VRTs are failing here due to test instability which is being resolved in #8402, after this is merged we can update this branch and re-run checks.

File: includes/Core/Modules/Modules.php [hunk 1]
```diff
@@ -177,9 +176,7 @@ public function __construct(
 		$this->authentication   = $authentication ?: new Authentication( $this->context, $this->options, $this->user_options );
 		$this->assets           = $assets ?: new Assets( $this->context );
 
-		if ( Feature_Flags::enabled( 'adsModule' ) ) {
-			$this->core_modules[ Ads::MODULE_SLUG ] = Ads::class;
-		}
```
Comment (eugene-manuilov) [inline includes/Core/Modules/Modules.php:182]: Let's add the Ads module to the core modules directly, similarly how we do for other modules: https://github.com/google/site-kit-wp/blob/ee48119f96675c792cb7a556edcdccad7a1f435b/includes/Core/Modules/Modules.php#L146-L153

**Review by eugene-manuilov — CHANGES_REQUESTED** [review]
Thanks, @benbowler. Mostly looks good to me. Just one comment from me.

**Comment by benbowler** [comment]
Thanks @eugene-manuilov, I've addressed your comment.

**Review by eugene-manuilov — APPROVED** [review]

**Comment by eugene-manuilov** [comment]
@benbowler, could you please resolve merge conflicts?

**Pull request merged**